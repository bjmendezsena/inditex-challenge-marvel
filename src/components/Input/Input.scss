@import "@/styles/variables.scss";

$width: 100%;

.container {
  position: relative;
  width: $width;
  display: flex;
  padding: 5px;
  &__input {
    background: none;
    color: $black-color;
    font-size: 18px;
    padding: 10px 10px 10px 5px;
    display: block;
    width: $width;
    border: none;
    border-radius: 0;
    height: 20px;

    &:focus {
      outline: none;
    }
  }

  &__bar {
    bottom: 0;
    position: absolute;
    transition: border-color 0.2s;
    display: block;
    width: $width;
    border-bottom: 2px solid $black-color;
  }
}
