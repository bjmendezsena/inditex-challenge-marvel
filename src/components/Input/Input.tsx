import { ComponentProps } from 'react';
import './Input.scss';

export type InputProps = {
  leftAddon?: React.ReactNode;
} & ComponentProps<'input'>;

export const Input = ({ leftAddon, ...props }: InputProps) => {
  return (
    <div className="container">
      {leftAddon}
      <input className="container__input" {...props} />
      <div className="container__bar" />
    </div>
  );
};
