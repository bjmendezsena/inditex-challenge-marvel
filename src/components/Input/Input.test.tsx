import { render } from '@testing-library/react';
import { Input } from './Input';

describe(`<${Input.name} />`, () => {
  it('should render without crashing', () => {
    const { asFragment } = render(<Input />);

    expect(asFragment()).toMatchSnapshot();
  });
  it('should render with addon left  without crashing', () => {
    const { asFragment } = render(<Input leftAddon={<>Left</>} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
