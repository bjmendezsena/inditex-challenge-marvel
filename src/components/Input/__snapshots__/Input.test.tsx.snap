// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Input /> > should render with addon left  without crashing 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    Left
    <input
      class="container__input"
    />
    <div
      class="container__bar"
    />
  </div>
</DocumentFragment>
`;

exports[`<Input /> > should render without crashing 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <input
      class="container__input"
    />
    <div
      class="container__bar"
    />
  </div>
</DocumentFragment>
`;

exports[`<Input /> should render with addon left  without crashing 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    Left
    <input
      class="container__input"
    />
    <div
      class="container__bar"
    />
  </div>
</DocumentFragment>
`;

exports[`<Input /> should render without crashing 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <input
      class="container__input"
    />
    <div
      class="container__bar"
    />
  </div>
</DocumentFragment>
`;
